engine/scripts.py

```diff
@@ -37,6 +37,14 @@ class KeyPoint:
     properties: Dict[str, Any]
 
 
+@dataclasses.dataclass
+class Message:
```

Comment:
Better to move this into `messages.py`. In general you'd only expose a type if things need it; it's not used in any protocol so no need to put it in here.